x/lockup/keeper/lock.go

```diff
@@ -435,6 +509,88 @@ func (k Keeper) ResetAllLocks(ctx sdk.Context, locks []types.PeriodLock) error {
 	return nil
 }
 
+func (k Keeper) ResetAllSyntheticLocks(ctx sdk.Context, syntheticLocks []types.SyntheticLock) error {
```

Comment:
Can we re-use any of this logic with normal Locks?

Reply:
Synthetic locks are using different entries to avoid conflicts with native lockups and I think we will need to keep this initialization logic separately from native lockups.

Reply:
I think we can re-use all of the accumulation store logic at least. Doesn't need to block this PR though, we can make this into a separate issue and do it later.

Reply:
Ok